src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.TaggerProvider.cs

```diff
@@ -124,8 +124,7 @@ private void ProduceTags(TaggerContext<TTag> context, DocumentSnapshotSpan spanT
                 {
                     if (_owner.IncludeDiagnostic(diagnosticData))
                     {
-                        var actualSpan = diagnosticData
-                            .GetExistingOrCalculatedTextSpan(sourceText)
+                        var actualSpan = AdjustSpan(diagnosticData.GetExistingOrCalculatedTextSpan(sourceText), sourceText)
```

Comment:
We should only do this if this is the old diagnostics.  Otherwise, this will mask issues if we have busted "live diagnostics".  We should have a bool that we pass along to indicate if the diagnostics are stale or are live.